compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -960,7 +976,7 @@ class RefChecks extends MiniPhase { thisPhase =>
   override def transformTemplate(tree: Template)(implicit ctx: Context) = try {
     val cls = ctx.owner
     checkOverloadedRestrictions(cls)
-    checkParents(cls)
+    checkParents(cls, tree)
```

Comment:
I'd use two different methods for checking parents:

    checkParent(cls)
    if (cls.is(Trait)) tree.parents.foreach(checkParentPrefix(cls, _))

That way the logic of `checkParents` can be left unchanged